packages/react/src/components/Masthead/Masthead.js

```diff
@@ -5,38 +5,173 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import PropTypes from 'prop-types';
 import React from 'react';
+import PropTypes from 'prop-types';
 import MastheadSearch from './MastheadSearch';
+import { settings } from 'carbon-components';
+import HeaderContainer from 'carbon-components-react/lib/components/UIShell/HeaderContainer';
+import UserProfile20 from '@carbon/icons-react/lib/user--profile/20';
+import { ReactComponent as Logo } from '../Icon/svg/ibm-logo.svg';
+import {
+  Header,
+  HeaderMenuButton,
+  HeaderGlobalBar,
+  HeaderGlobalAction,
+  HeaderName,
+  HeaderNavigation,
+  HeaderMenu,
+  HeaderMenuItem,
+  HeaderSideNavItems,
+  SkipToContent,
+  SideNav,
+  SideNavItems,
+  SideNavLink,
+  SideNavMenu,
+  SideNavMenuItem,
+} from 'carbon-components-react/lib/components/UIShell';
+import '@ibmdotcom/styles/scss/components/masthead/_masthead.scss';
+
+const { prefix } = settings;
 
 /**
- * Test MastHead component
+ * MastHead component
  *
+ * @typedef {object} navigation Object containing navigation elements
  * @param {string} type Type of masthead
  * @returns {*} Masthead component
- * @class
  */
-const Masthead = ({ type }) => {
-  if (type === 'branded') {
-    return (
-      <div className="masthead">
-        <MastheadSearch />
-      </div>
-    );
-  } else {
-    return (
-      <div className="masthead">
-        <MastheadSearch />
-      </div>
-    );
-  }
+const Masthead = ({ navigation }) => {
+  const navigationLinks = navigation.links;
+
+  return (
+    <HeaderContainer
+      render={({ isSideNavExpanded, onClickSideNavExpand }) => (
+        <>
+          <Header aria-label="IBM" data-autoid={`${prefix}--masthead`}>
+            <SkipToContent />
+            <HeaderMenuButton
+              aria-label="Open menu"
+              data-autoid={`${prefix}--masthead__hamburger`}
+              onClick={onClickSideNavExpand}
+              isActive={isSideNavExpanded}
+            />
+            {navigation.platform ? (
+              <>
+                <div
+                  className={`${prefix}--header__logo ${prefix}--header__logo--platform`}>
+                  <a href="https://ibm.com">
+                    <Logo data-autoid={`${prefix}--masthead__logo`} />
```

Comment:
This is something I actually meant to ask you guys but forgot. `--header__` is what Carbon uses. I started using `--masthead__` because that's what we've been calling it and I guess it was in my head when I started typing. Should we standardize and use `header` throughout?

cc @jeffchew

Reply:
I lean towards staying consistent with `--header__`. If they have both of our headers on the same page they are doing some definitely wrong.